### PR TITLE
Implement basic Galley metadata code / infrastructure for K8s

### PR DESCRIPTION
**Pull request by ozevren**
This PR implements basic metadata/proto building blocks and the core infrastructure for reading from K8s.

### CODE REVIEW

File: galley/pkg/kube/info.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	sc "k8s.io/apimachinery/pkg/runtime/schema"
+
+	"istio.io/istio/galley/pkg/runtime/resource"
+)
+
+// Info represents a known crd. It is used to drive the K8s-related machinery, and to map to
+// the proto format.
+type Info struct {
```
Comment (ayj) [inline galley/pkg/kube/info.go]: `kube.Info` is pretty generic. `Schema`, `ResourceSpec`, etc. would be more descriptive.

File: galley/pkg/kube/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+// NewKube returns a new instance of Interfaces.
+func NewKube(cfg *rest.Config) Interfaces {
+	return &kube{
+		cfg: cfg,
+	}
+}
+
+// DynamicInterface returns a new dynamic.Interface for the specified API Group/Version.
+func (k *kube) DynamicInterface(gv schema.GroupVersion, kind, listKind string) (
```
Comment (ayj) [inline galley/pkg/kube/interfaces.go]: My previous comment about named returned parameters also applies here.

**Comment by ozevren** [comment]
Updated

File: galley/pkg/kube/interfaces.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+// NewKube returns a new instance of Interfaces.
+func NewKube(cfg *rest.Config) Interfaces {
+	return &kube{
+		cfg: cfg,
+	}
+}
+
+// DynamicInterface returns a new dynamic.Interface for the specified API Group/Version.
+func (k *kube) DynamicInterface(gv schema.GroupVersion, kind, listKind string) (dynamic.Interface, error) {
+	var result dynamic.Interface
```
Comment (ayj) [inline galley/pkg/kube/interfaces.go]: The forward declaration isn't needed. Something like the following would be more idiomatic. 

```golang
cfg, err := k.createConfig(gv, kind, listKind)
if err != nil {
    return nil, err
}
return dynamic.NewClient(cfg)
````

Comment (ozevren) [reply galley/pkg/kube/interfaces.go]: That leaves hard to code-coverage code blocks. :)

**Pull request merged**